app/controllers/concerns/api/import_puppetclasses_common_controller.rb

```diff
@@ -2,9 +2,15 @@ module Api::ImportPuppetclassesCommonController
   extend ActiveSupport::Concern
 
   included do
-    before_action :find_required_puppet_proxy, :only => [:import_puppetclasses]
-    before_action :get_environment_id, :only => [:import_puppetclasses]
-    before_action :find_optional_environment, :only => [:import_puppetclasses]
+    prepend_before_action :fail_and_inform_about_plugin_for_import, only: :import_puppetclasses
+  end
+
+  def resource_human_name
+    resource_class.model_name.name
+  end
+
+  def fail_and_inform_about_plugin
+    render json: { message: _('To access import_puppetclasses API you need to install Foreman Puppet Enc plugin')}, status: :not_implemented
```

Comment:
Please keep naming consistent. https://github.com/theforeman/foreman_puppet calls it Foreman Puppet.
```suggestion
    render json: { message: _('To access import_puppetclasses API you need to install the Foreman Puppet plugin')}, status: :not_implemented
```

Reply:
Thanks for noticing! The renaming got messy ;)